lms/djangoapps/course_api/blocks/transformers/video_urls.py

```diff
@@ -46,7 +46,11 @@ def transform(self, usage_info, block_structure):
             )
             if not student_view_data:
                 return
-            encoded_videos = student_view_data['encoded_videos']
+
+            # encoded_videos key doesn't exist for web-only videos
+            encoded_videos = student_view_data.get('encoded_videos', False)
+            if not encoded_videos:
+                continue
```

Comment:
If the key doesn't exist, we will continue the work on the rest of the videos. However, if we look at the L47-L48, when the student_view_data is not found, the function is being stopped with the return. This is intentional as student_view_data is a query param to our course blocks API and it returns student_view_data for a particular block type. For example, to get the video data, we specify **student_view_data=video** in the URL. We can't get student_view_data for some video only(as far as I know). It is all or none. So, if there is no student_view_data, there is no need to check for the remaining videos and end the function there. I thought to mention this bit, just in case anyone wonders the choice of two statements.